Core/src/Surfaces/Surface.cpp

```diff
@@ -357,3 +357,16 @@ void Acts::Surface::assignSurfaceMaterial(
 void Acts::Surface::associateLayer(const Acts::Layer& lay) {
   m_associatedLayer = (&lay);
 }
+
+std::ostream& Acts::operator<<(std::ostream& os, const Acts::Surface& srf) {
+  if (srf.associatedDetectorElement() != nullptr) {
+    throw std::runtime_error(
+        "Cannot print Surface with associated DetectorElement without geometry "
+        "context");
+  }
+  // Using a default context is ONLY safe, if the surface does not have an
+  // associated detector element, which we ensure right above.
+  Acts::GeometryContext defaultContext;
+  os << srf.toStream(defaultContext);
```

Comment:
I wonder if there should also be a `toStream` overload without context which then drives the exception also allowing for overrides later

Reply:
This requires implementing it on all derived surface types. I'm not sure that's needed, and they would likely also call each other internally to avoid copying code. Does that make sense?

Reply:
I thought we could default it to your behavior here and give the surface the opportunity to specialize but maybe this does not have any use case anyways

Reply:
I see your point. I guess the only reason using `toStream` over `operator<<` is that with `toStream` you can in fact pass a geometry context.